test/test_jit.py

```diff
@@ -375,7 +375,9 @@ def test_backward(self):
         y = a * b
 
         trace = torch._C._tracer_enter((x, y), 2)
-        z = y * 2 * x
+        # NB: Parenthesization here seems important, otherwise the
+        # evaluation order is nondeterministic
+        z = (y * 2) * x
```

Comment:
Whaaat?

Reply:
OK, never mind, I reread the manual rules on evaluation order https://docs.python.org/3/reference/expressions.html#evaluation-order and I am not sure how the test failure I cited here can ever occur... Python bug, maybe?

Reply:
Nah, I just misinterpreted the diff:

```
  ---------------- stage 1 ----------------
-   %6 : Double(2, 2) = mul(%5, %1), uses = [%8.i0];
-   %7 : Double(2, 2) = mul(%5, %3), uses = [%0.i1];
?    ^
+   %6 : Double(2, 2) = mul(%5, %3), uses = [%0.i1];
?    ^
+   %7 : Double(2, 2) = mul(%5, %1), uses = [%8.i0];
-   %8 : Double(2, 2) = mul[other={2}](%6), uses = [%0.i2];
?                                       ^
+   %8 : Double(2, 2) = mul[other={2}](%7), uses = [%0.i2];
?                                       ^
```

Stage 0 is changed, but only the uses. Never mind, it's the same situation as the other case.